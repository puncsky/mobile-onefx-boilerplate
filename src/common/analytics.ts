import { Analytics, Event, PageHit } from "expo-analytics";
// @ts-ignore
import ExpoMixpanelAnalytics from "expo-mixpanel-analytics";
import { config } from "../config";

const ga = __DEV__
  ? // tslint:disable-next-line:no-any
    ({} as any)
  : new Analytics(config.analytics.googleTid, undefined, {
      debug: __DEV__
    });

const mixpanel = __DEV__
  ? // tslint:disable-next-line:no-any
    ({} as any)
  : new ExpoMixpanelAnalytics(config.analytics.mixpanelProjectToken);

const analytics = {
  identify(id: string): void {
    if (__DEV__) {
      return;
    }

    mixpanel.identify(id);
    // @ts-ignore
    ga.parameters.uid = id;
  },

  // tslint:disable-next-line:no-any
  async track(name: string, props: Record<string, any>): Promise<void> {
    if (__DEV__) {
      return;
    }

    mixpanel.track(name, props);
    if (name.startsWith("page_view_")) {
      await ga.hit(new PageHit(name));
    } else if (name.startsWith("tap_")) {
      await ga.event(new Event("tap", name, props.id));
    } else {
      await ga.event(new Event(name, name));
    }
  },

  people_delete_user(): void {
    mixpanel.people_delete_user();
  }
};

export { analytics };
