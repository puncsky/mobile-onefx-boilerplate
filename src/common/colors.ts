export const colors = {
  primary: "#FBB03B",
  secondary: "#0C8DE4",
  white: "#fff",

  black: "#000000",
  black90: "#333333",
  black80: "#999999",
  black60: "#CCCCCC",
  black40: "#E5E5E5",
  black20: "#F0F0F0",
  black10: "#F7F7F7",

  text01: "#4c4c4c", //		Primary text, Body copy

  error: "#E54937", //	Error
  success: "#07A35A", //	Success
  warning: "#FFA000", //	Warning
  information: "#5aaafa", //	Information

  nav01: "#011627", //	Global top bar
  nav02: "#20232a", //	CTA footer

  tabIconDefault: "#ccc",
  tabIconSelected: "#2f95dc"
};

export const antdTheme = {
  brand_primary: colors.primary,
  color_link: colors.primary,
  primary_button_fill: colors.primary,
  primary_button_fill_tap: colors.primary
};
