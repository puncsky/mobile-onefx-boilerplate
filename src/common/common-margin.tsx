import { View } from "react-native";
import React from "react";

export function CommonMargin(): JSX.Element {
  return <View style={{ margin: 8 }} />;
}
