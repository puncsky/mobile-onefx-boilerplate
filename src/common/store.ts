import * as Locatization from "expo-localization";
import { applyMiddleware, createStore } from "redux";
import { composeWithDevTools } from "redux-devtools-extension";
import { persistStore } from "redux-persist";
import thunk from "redux-thunk";
import { persistedReducer } from "./root-reducer";
import { theme } from "./theme";

export interface AppState {
  base: {
    userId?: string;
    authToken?: string;
    mixpanelId?: string;
    locale?: string;
    currentTheme?: "dark" | "light";
  };
}

type Mode = "dark" | "light";

const preloadedState: AppState = {
  base: {
    locale: Locatization.locale,
    currentTheme: theme.name as Mode
  }
};

const middleware = [thunk];

export const store = createStore(
  persistedReducer,
  preloadedState as any,
  composeWithDevTools(applyMiddleware(...middleware))
);

export const persistor = persistStore(store);
