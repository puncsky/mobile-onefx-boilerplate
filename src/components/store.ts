import { createStore } from "redux";

export interface AppState {}

export interface Action {
  type: string;
  payload: Object;
}

const INITIAL_STATE: AppState = {
  snippets: []
};

const UPDATE_REDUX_STATE = "UPDATE_REDUX_STATE";

const reducer = (state = INITIAL_STATE, action: Action) => {
  if (action.type === UPDATE_REDUX_STATE) {
    return {
      ...state,
      ...action.payload
    };
  }
  return state;
};

export const store = createStore(reducer, {});
export const actionUpdateReduxState = (payload: Object) => ({
  type: UPDATE_REDUX_STATE,
  payload
});
