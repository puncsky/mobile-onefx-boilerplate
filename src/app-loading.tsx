//@ts-ignore
import * as Icon from "@expo/vector-icons";
import { AppLoading } from "expo";
import { Asset } from "expo-asset";
import * as Font from "expo-font";
import * as React from "react";
import { connect } from "react-redux";
import { Dispatch } from "redux";
import { Action, actionUpdateReduxState } from "./common/root-reducer";
import { AppState } from "./common/store";
import { theme } from "./common/theme";
import i18n from "./translations";
import { ThemeProps } from "./types/theme-props";

export function AppLoaderRoot({ onFinish }: { onFinish(): void }): JSX.Element {
  //@ts-ignore
  return <AppLoadingContainer onFinish={onFinish} />;
}

interface Props {
  actionUpdateReduxState(payload: object): { type: string; payload: object };
  onFinish(): void;
  mixpanelId?: string;
  userId?: string;
  locale?: string;
  currentTheme?: ThemeProps;
}

const AppLoadingContainer = connect<
  {},
  (
    dispatch: Dispatch<Action>
  ) => { actionUpdateReduxState(payload: Object): void },
  { onFinish(): void }
>(
  (state: AppState) => ({
    userId: state.base.userId,
    mixpanelId: state.base.mixpanelId,
    locale: state.base.locale,
    currentTheme: state.base.currentTheme
  }),
  dispatch => ({
    actionUpdateReduxState(payload: Object): void {
      dispatch(actionUpdateReduxState(payload));
    }
  })
)(function AppLoadingInner(props: Props): JSX.Element {
  const { locale, onFinish, currentTheme, actionUpdateReduxState } = props;

  if (locale) {
    i18n.locale = locale;
  }

  if (!currentTheme) {
    actionUpdateReduxState({
      base: {
        currentTheme: theme.light
      }
    });
  }

  const loadResourcesAsync = async () => {
    try {
      await Promise.all([
        Asset.loadAsync([
          require("./assets/images/robot-dev.png"),
          require("./assets/images/robot-prod.png")
        ]),
        Font.loadAsync({
          // This is the font that we are using for our tab bar
          ...Icon.Ionicons.font,
          // We include SpaceMono because we use it in HomeScreen.js. Feel free
          // to remove this if you are not using it in your app
          "space-mono": require("./assets/fonts/SpaceMono-Regular.ttf")
        }),
        Font.loadAsync(
          "antoutline",
          require("../node_modules/@ant-design/icons-react-native/fonts/antoutline.ttf")
        )
      ]);
    } catch (error) {
      // tslint:disable-next-line
      console.error(`failed to loadResourcesAsync: ${error}`);
    }
  };

  const handleLoadingError = (error: Error) => {
    // In this case, you might want to report the error to your error
    // reporting service, for example Sentry
    console.warn(error);
  };

  return (
    <AppLoading
      startAsync={loadResourcesAsync}
      onError={handleLoadingError}
      onFinish={onFinish}
      autoHideSplash={false}
    />
  );
});
