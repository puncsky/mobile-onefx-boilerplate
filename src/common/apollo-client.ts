import { InMemoryCache } from "apollo-cache-inmemory";
import { ApolloClient } from "apollo-client";
import { ApolloLink } from "apollo-link";
import { HttpLink } from "apollo-link-http";
import fetch from "isomorphic-unfetch";
import { store } from "./store";

export const ENDPOINT = "https://guigu.io/";

export const headers: { [key: string]: string } = {
  "x-app-id": "mobile-guiguio"
};

export const getEndpoint = (path: string) => `${ENDPOINT}${path}`;

const middlewareLink = new ApolloLink((operation, forward) => {
  operation.setContext({
    headers: { authorization: `Bearer ${store.getState().base.authToken}` }
  });
  return forward(operation);
});

// use with apollo-client
const link = middlewareLink.concat(
  new HttpLink({
    uri: getEndpoint("api-gateway/"),
    fetch
  })
);

export const apolloClient = new ApolloClient({
  link,
  cache: new InMemoryCache()
});
