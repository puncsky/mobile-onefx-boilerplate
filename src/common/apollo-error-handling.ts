import { onError } from "apollo-link-error";
import { store } from "@/common/store";

export const onErrorLink = onError(({ graphQLErrors, networkError }) => {
  if (graphQLErrors) {
    for (const err of graphQLErrors) {
      if (err.extensions && err.extensions.code === "UNAUTHENTICATED") {
        store.dispatch({
          type: "UPDATE_REDUX_STATE",
          payload: {
            base: {
              userId: "",
              authToken: ""
            }
          }
        });
      }
    }
  }
  if (networkError) {
    // eslint-disable-next-line no-console
    console.log(`[Network error]: ${networkError}`);
  }
});
