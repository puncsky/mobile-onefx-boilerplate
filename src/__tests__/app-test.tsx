import * as React from "react";
import "react-native";
import { StatusBar } from "react-native";
// @ts-ignore
import NavigationTestUtils from "react-navigation/NavigationTestUtils";
import { Provider } from "react-redux";
import * as renderer from "react-test-renderer";
import { AppLoaderRoot } from "../app-loading";
import { store } from "../common/store";

describe("App snapshot", () => {
  beforeEach(() => {
    jest.useFakeTimers();
    NavigationTestUtils.resetInternalState();
  });

  it("renders the StatusBar", async () => {
    const tree = renderer.create(<StatusBar />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders AppLoaderRoot", async () => {
    const tree = renderer
      .create(
        /* tslint:disable:no-any */
        <Provider store={store as any}>
          <AppLoaderRoot onFinish={() => null} />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

// import * as React from "react";
// import "react-native";
// import { View } from "react-native";
// // @ts-ignore
// import NavigationTestUtils from "react-navigation/NavigationTestUtils";
// import * as renderer from "react-test-renderer";
//
// describe("App snapshot", () => {
//   beforeEach(() => {
//     jest.useFakeTimers();
//     NavigationTestUtils.resetInternalState();
//   });
//
//   it("renders the loading screen", async () => {
//     const tree = renderer.create(<View />).toJSON();
//     expect(tree).toMatchSnapshot();
//   });
// });
