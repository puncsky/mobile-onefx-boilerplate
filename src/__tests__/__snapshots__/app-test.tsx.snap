// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot renders AppLoaderRoot 1`] = `null`;

exports[`App snapshot renders the StatusBar 1`] = `null`;
